### PR TITLE
release to maven central and push uber jar to github releases on release

### LINKED ISSUE #64: Handle releases correctly
If we create a new tag defining a version on the spark-api repository, we should:
- Execute all the tests
- Generate a `uber-jar` using `sbt assembly`
- Generate a new container version including the new `uber-jar`
- Upload assembly jar to spark packages
- Upload python code to pypi

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #64 

I also removed the `openjdk8` from the main build and added it as a matrix include to be able to override the deploy phase so it does not try to deploy more than one time for each jdk.

Also rearranged the travis file to avoid having to redefine some stuff in other jobs of the matrix

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -41,3 +41,58 @@ assemblyMergeStrategy in assembly := {
...
+developers += Developer("ajnavarro",
+  "Antonio Navarro",
+  "antonio@sourced.tech",
+  url("https://github.com/ajnavarro"))
+developers += Developer("bzz",
+  "Alexander Bezzubov",
+  "alex@sourced.tech",
+  url("https://github.com/bzz"))
+developers += Developer("mcarmonaa",
+  "Manuel Carmina",
```
Comment (mcarmonaa) [inline build.sbt]: s/Carmina/Carmona

File: .travis.yml [hunk 2]
```diff
@@ -79,10 +84,21 @@ script:
...
+  - provider: script
+    script: make docker-push && make maven-release
+    skip_cleanup: true
+    on:
+      tags: true
+  - provider: releases
+    api_key:
+      secure: $GITHUB_TOKEN
+    file_glob: true
+    file: *.jar
```
Comment (mcarmonaa) [inline .travis.yml]: Does travis expand this  `*.jar`?

Comment (erizocosmico) [reply .travis.yml]: Yup, that's what the `file_glob: true` is for

File: build.sbt (hunk 1 above)
Comment (erizocosmico) [reply build.sbt]: :fearful:

**Comment by erizocosmico** [comment]
Fixed @mcarmonaa

**Comment by erizocosmico** [comment]
Build failing because of jitpack and siva java, don't merge til the build is run

**Comment by erizocosmico** [comment]
Build fixed, now it can be merged :shipit:

**Pull request merged**